diff --git a/config.def.h b/config.def.h
index e06be66..c93abe1 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,17 +1,17 @@
 /* See LICENSE file for copyright and license details. */
 
 /* interval between updates (in ms) */
-const unsigned int interval = 1000;
+const unsigned int interval = 3000;
 
 /* text to show if no value can be retrieved */
-static const char unknown_str[] = "n/a";
+static const char unknown_str[] = "No Signal";
 
 /* maximum output string length */
 #define MAXLEN 2048
 
 /*
  * function            description                     argument (example)
- *
+ 
  * battery_perc        battery percentage              battery name (BAT0)
  *                                                     NULL on OpenBSD/FreeBSD
  * battery_state       battery charging state          battery name (BAT0)
@@ -61,7 +61,11 @@ static const char unknown_str[] = "n/a";
  * wifi_perc           WiFi signal in percent          interface name (wlan0)
  * wifi_essid          WiFi ESSID                      interface name (wlan0)
  */
+
 static const struct arg args[] = {
-	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+	/* function          format          argument */
+
+    { wifi_essid,        "  %s   ",    "wlp1s0"},
+	{ battery_perc,      "  %s%%   ",  "BAT0"  },
+    { datetime,          "  %s 神",     "%H:%M" },
 };
